s3stream/src/main/java/com/automq/stream/s3/operator/DefaultS3Operator.java

```diff
@@ -442,24 +444,73 @@ public CompletableFuture<Void> delete(String path) {
-            .build();
-        // TODO: handle not exist object, should we regard it as deleted or ignore it.
-        return this.writeS3Client.deleteObjects(request).thenApply(resp -> {
-            S3OperationStats.getInstance().deleteObjectsStats(true).record(timerUtil.elapsedAs(TimeUnit.NANOSECONDS));
-            LOGGER.info("[ControllerS3Operator]: Delete objects finished, count: {}, cost: {}", resp.deleted().size(), timerUtil.elapsedAs(TimeUnit.NANOSECONDS));
-            return resp.deleted().stream().map(DeletedObject::key).collect(Collectors.toList());
-        }).exceptionally(ex -> {
-            S3OperationStats.getInstance().deleteObjectsStats(false).record(timerUtil.elapsedAs(TimeUnit.NANOSECONDS));
-            LOGGER.info("[ControllerS3Operator]: Delete objects failed, count: {}, cost: {}, ex: {}", objectKeys.size(), timerUtil.elapsedAs(TimeUnit.NANOSECONDS), ex.getMessage());
-            return Collections.emptyList();
-        });
+                .bucket(bucket)
+                .delete(Delete.builder()
+                        .objects(toDeleteKeys)
+                        .quiet(true)
+                        .build())
+                .build();
+
+        return this.writeS3Client.deleteObjects(request)
+                .thenApply(resp -> handleDeleteObjectsResponse(objectKeys, timerUtil, resp))
+                .exceptionally(ex -> {
+                    S3OperationStats.getInstance().deleteObjectsStats(false)
+                            .record(timerUtil.elapsedAs(TimeUnit.NANOSECONDS));
+                    LOGGER.info("[ControllerS3Operator]: Delete objects failed, count: {}, cost: {}, ex: {}",
+                            objectKeys.size(), timerUtil.elapsedAs(TimeUnit.NANOSECONDS), ex.getMessage());
+                    return Collections.emptyList();
+                });
+    }
+
+    static List<String> handleDeleteObjectsResponse(List<String> objectKeys,
+                                                    TimerUtil timerUtil,
+                                                    DeleteObjectsResponse response) {
+        // quiet mode won't return any success key. so think as all success.
+        Set<String> successDeleteKeys = new HashSet<>(objectKeys);
+        int errDeleteCount = 0;
+        boolean hasUnExpectedResponse = false;
+
+        for (S3Error error : response.errors()) {
+            if ("NoSuchKey".equals(error.code())) {
+                // ignore for delete objects.
+                continue;
+            }
+
+            if (!StringUtils.isEmpty(error.key())) {
+                successDeleteKeys.remove(error.key());
+            } else {
+                hasUnExpectedResponse = true;
+            }
+
+            LOGGER.error("[ControllerS3Operator]: Delete objects for key [{}] error code [{}] message [{}]",
+                    error.key(), error.code(), error.message());
+
+            errDeleteCount++;
+        }
+
+        if (hasUnExpectedResponse) {
```

Comment:
Rely on the semantics of DeleteObjects. Both NoSuchKey and those not on the list are considered to be deleted successfully.